tests/components/with_search.spec.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+
+import ShallowRenderer from "react-test-renderer/shallow";
+import withSearch from "../../src/components/with_search.js";
+import SourceList from "../../src/components/source_list";
+
+const ListComponent = withSearch(SourceList);
+const messages = { searchPlaceholder: "Search" };
+const UserSearch = props => (
+  <input type="text" placeholder={messages.searchPlaceholder} {...props} />
+);
+
+const testParameters = {
+  selectAllRenderer: true,
+  noItemsRenderer: false,
+  showSelectAll: true,
+  selectItem: () => {},
+  selectAllItems: () => {},
+  itemHeight: 50,
+  isAllSelected: false,
+  itemRenderer: () => <span>Test</span>,
+  filterItems: () => {},
+  selectAllHeight: 400,
+  calculatedHeight: 50,
+  filteredItems: [
+    { id: 1, label: "item-1", disabled: true },
+    { id: 2, label: "item-2", disabled: false },
+    { id: 3, label: "item-3", withGrouping: true },
+    { id: 4, label: "item-4" }
+  ],
+  selectedIds: [1, 2],
+  getList: () => {},
+  listRenderer: () => {}
+};
+
+describe("With search", () => {
```

Comment:
These tests are very coupled with SourceList.
This means that:
1. if SourceList changes it will change these tests from no reason.
2. withSearch was supposed to work on DestinationList as well, I see not reason to keep the test limited to SourceList or the two components.

Consider testing withSearch features alone without having the test know about the props the list expects to pass.
This will greatly simplify the test as well as make it more robust.